### PR TITLE
Add random events (framework + events)

### PR DESCRIPTION
**Pull request by HarleyGilpin**
## Overview

Implements the anti-macro **Random Events** D&D (ref GregHib/void-private#223). This is the main PR for the feature: it lands the shared framework and then accumulates each of the 18 current (2011) random events as they're built, verified against the wiki (`oldid=3667851`).

## Framework

- **Lifecycle** (`content/activity/event/random/RandomEvents`): weighted `pick()` over the `random_events` table, `start`/`complete`/`fail`, persisted per-player cooldown, and `noteAndTeleport` (ignore penalty — notes the inventory then exiles the player so an event can't be macroed through).
- **Trigger** (`RandomEventTrigger`): rolls per XP drop (1-in-N), gated by an `eligible()` check (no bots, combat, instances, dialogue, or `no_random_events` areas), plus a `::randomevent [event]` admin command.
- **Entry** (`RandomEventKidnap`): relog resume, teleport-out blocking, the `kidnap(tile)` helper, and the `mysteriousOldMan()` intro. Two supported patterns:
  - *Kidnap-to-instance/room* — the player is teleported to an event area (Maze, Freaky Forester).
  - *In-place nagging* (`InPlaceRandomEvent`) — the event NPC spawns beside the player, follows and nags, and applies the ignore penalty on timeout (Certer, Sandwich Lady, Evil Twin...).

## Extras
- Add Iffie's costume store 
- Add random event gift rewards interface
- Add XP lamps

## Media / Screenshots
image

image

## Events

- [x] **Maze** — GregHib/void-private#257
- [x] **Freaky Forester** — GregHib/void-private#255
- [x] **Certer** — GregHib/void-private#253
- [x] **Drill Demon** — GregHib/void-private#254
- [x] **Kiss the Frog** — GregHib/void-private#256
- [x] **Mime** — GregHib/void-private#258
- [x] Cap'n Arnav's Chest — GregHib/void-private#259
- [x] **Quiz Master** — GregHib/void-private#260
- [x] **Sandwich Lady** — GregHib/void-private#261
- [x] **Evil Bob** — GregHib/void-private#262
- [x] **Surprise Exam** — GregHib/void-private#263
- [x] Bee-Keeper — GregHib/void-private#264
- [x] Evil Twin — GregHib/void-private#265
- [x] Pinball — GregHib/void-private#266
- [x] Gravedigger — GregHib/void-private#267
- [x] **Pillory** — GregHib/void-private#268
- [x] Prison Pete — GregHib/void-private#269
- [x] Lost and Found — GregHib/void-private#270

## Manual Testing
- Manually verified each random event end to end.
[Screen recording](https://cdn.discordapp.com/attachments/1079992933526278154/1525278514616602865/Screencast_20260710_155312.webm?ex=6a52cdd0&is=6a517c50&hm=39dc7533cc204a7199a89351352fdd7f3edd3ce33e2e4b09eb954976570eb67d&)
## Testing

`./gradlew :game:test --tests "content.activity.event.random.*"` — framework + Maze + Freaky Forester suites green. Spotless clean.

## Manual test

`::random_event <event>` starts any registered event (or `::random_event` for a random pick). Current: `maze`, `freaky_forester`.

### CODE REVIEW

File: game/src/main/kotlin/content/activity/event/random/RandomEvents.kt [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+ * [complete] when the player succeeds or [fail] when they fail or abandon it.
+ */
+object RandomEvents {
+
+    private val events = mutableMapOf<String, suspend Player.() -> Unit>()
+
+    fun register(id: String, launcher: suspend Player.() -> Unit) {
+        Script.checkLoading()
+        events[id] = launcher
+    }
```
Comment (GregHib) [inline game/src/main/kotlin/content/activity/event/random/RandomEvents.kt:45]: It probably doesn't matter as you're using a map with unique keys but it would be good practise to have a clear() method like other Script api's because script .init's and registrations can be called more than once (like in a script reload, or tests) which could lead to duplicates or invalid states.

File: game/src/main/kotlin/content/activity/event/random/RandomEventRewards.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            return
+        }
+    }
+}
+
+private fun Player.giveOrDrop(item: String, amount: Int) {
+    if (!inventory.add(item, amount)) {
+        FloorItems.add(tile, item, amount, disappearTicks = 300, owner = this)
+    }
+}
```
Comment (GregHib) [inline game/src/main/kotlin/content/activity/event/random/RandomEventRewards.kt]: addOrDrop already exists

File: game/src/main/kotlin/content/activity/event/random/RandomEvents.kt (hunk 1 above)
Comment (HarleyGilpin) [reply game/src/main/kotlin/content/activity/event/random/RandomEvents.kt:45]: Good call - made `RandomEvents` `AutoCloseable` (clears the launcher map) and registered it in `ContentLoader` alongside `PrayerApi`, so a reload resets it. https://github.com/GregHib/void/pull/1064/commits/023822c4be38d05c6e557d0920dc0c7fd3317ec3

File: game/src/main/kotlin/content/activity/event/random/RandomEventRewards.kt (hunk 2 above)
Comment (HarleyGilpin) [reply game/src/main/kotlin/content/activity/event/random/RandomEventRewards.kt]: Thanks - dropped the duplicated helpers and switched to `Player.addOrDrop` throughout the random events. https://github.com/GregHib/void/pull/1064/commits/023822c4be38d05c6e557d0920dc0c7fd3317ec3

**Review by GregHib — APPROVED** [review]
Just tidied up some settings, added some more exception areas and changed the trigger from xp to a timer-based, based off how it works in osrs.

One nit: overuse of passing npc ids in dialogues instead of using talkWith(npc) i.e. `npc<Happy>("npc_id", "text")` I think it's too risky/not worth the time to change them all now but for future ref.
